launch/launch/actions/execute_process.py

```diff
@@ -681,7 +686,19 @@ def __expand_substitutions(self, context):
         cmd = [perform_substitutions(context, x) for x in self.__cmd]
         name = os.path.basename(cmd[0]) if self.__name is None \
             else perform_substitutions(context, self.__name)
-        cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
+
+        # Peform filtering for prefix application
+        filter_str = perform_substitutions(context, self.__prefix_filter)
+        regex_filter = None
+        if len(filter_str):
+            regex_filter = re.compile(filter_str)
+        if regex_filter is not None and regex_filter.match(os.path.basename(cmd[0])):
+            cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
+
+        # Apply to all if no filter is provided or filter is overridden
+        if regex_filter is None or self.__override_filter:
+            cmd = shlex.split(perform_substitutions(context, self.__prefix)) + cmd
```

Comment:
@camm73 meta: hmm, if I read this correctly the prefix filter is irrelevant when a prefix is explicitly given. Nevermind the potential code simplification, that's actually a bit counter-intuitive when attempting to target an specific executable.

Reply:
My thinking behind this was based on the fact that whenever a user sets a specific prefix in their launch file it currently overrides any `launch-prefix` provided in the launch configuration and applies the specific prefix: https://github.com/ros2/launch/blob/f891aed9f904df6397ef554f7e0b36bb37b30529/launch/launch/actions/execute_process.py#L230-L232

Since the specific prefix already overrides a prefix passed via `--launch-prefix`, I figured it should also do the same when `--launch-prefix` and `--launch-prefix-filter` are used in conjunction, for the sake of consistency.

If you think it's more intuitive to remove the `override_filter` feature though, I can definitely do that.

Reply:
That's fair. Should be fine so long as we document it.

Reply:
Specifically, let's document this behavior in `ExecuteProcess.__init__()` docstring.